tests/replication2.erl

```diff
@@ -11,592 +11,692 @@
+  ClusterASize = rt_config:get(cluster_a_size, 3),
 
-    %% make sure we are connected
-    lager:info("Wait for cluster connection A:~p -> B:~p:~p", [LeaderA, BFirst, Port]),
-    ?assertEqual(ok, repl_util:wait_for_connection(LeaderA, "B")),
-
-    log_to_nodes(AllNodes, "Write data to A, verify replication to B via realtime"),
-    %% write some data on A
-    %io:format("~p~n", [rpc:call(LeaderA, riak_repl_console, status, [quiet])]),
-    lager:info("Writing 100 more keys to ~p", [LeaderA]),
-    ?assertEqual([], repl_util:do_write(LeaderA, 101, 200, TestBucket, 2)),
-
-    %% verify data is replicated to B
-    lager:info("Reading 100 keys written to ~p from ~p", [LeaderA, BFirst]),
-    ?assertEqual(0, repl_util:wait_for_reads(BFirst, 101, 200, TestBucket, 2)),
-
-    case Connected of
-        false ->
-            %% check that the keys we wrote initially aren't replicated yet, because
-            %% we've disabled fullsync_on_connect
-            lager:info("Check keys written before repl was connected are not present"),
-            Res2 = rt:systest_read(BFirst, 1, 100, TestBucket, 2),
-            ?assertEqual(100, length(Res2)),
-
-            log_to_nodes(AllNodes, "Test fullsync with leader ~p", [LeaderA]),
-            repl_util:start_and_wait_until_fullsync_complete(LeaderA),
-
-            case rpc:call(LeaderA, init, script_id, []) of
-                {"riak", Vsn} when Vsn > "1.4" ->
-                    %% check that the number of successful FS source exists matches the number of partitions
-                    NumExits = repl_util:get_fs_coord_status_item(LeaderA, "B", successful_exits),
-                    NumPartitions = repl_util:num_partitions(LeaderA),
-                    ?assertEqual(NumPartitions, NumExits);
-                _ ->
-                    ok
-            end,
-
-            lager:info("Check keys written before repl was connected are present"),
-            ?assertEqual(0, repl_util:wait_for_reads(BFirst, 1, 200, TestBucket, 2));
+  lager:info("Deploy ~p nodes", [NumNodes]),
+  Conf = [
+          {riak_kv,
+              [
+               %% Specify fast building of AAE trees
+               {anti_entropy, {on, []}},
+               {anti_entropy_build_limit, {100, 1000}},
+               {anti_entropy_concurrency, 100}
+              ]
+          },
+          {riak_repl,
+           [
+              {fullsync_on_connect, false},
+              {fullsync_interval, disabled},
+              {diff_batch_size, 10}
+           ]}
+  ],
+
+  lager:info( "Building Clusters A and B" ),
+  [ANodes, BNodes] = rt:build_clusters([{ClusterASize, Conf}, {NumNodes - ClusterASize, Conf}]),
+
+  %lager:info( "Skipping all tests" ),
+  replication(ANodes, BNodes, false),
+  pass.
+
+
+replication(ANodes, BNodes, Connected) ->
+
+  log_to_nodes(ANodes ++ BNodes, "Starting replication2 test"),
+  lager:info( "Connection Status: ~p", [Connected] ),
+
+  lager:info( "Real Time Full Sync Replication test" ),
```

Comment:
There has been very little function change in this refactoring.  Mainly, the flow of the original test was cascading in nature, with each test case picking up where the last one left off, reusing the variables from 3-5 test cases previous. To provide clarity, I broke the module down into sub tasks, and tried to reset the state of the Cluster after each test case.  In nearly all cases, any actual code changes were in the form of "LeaderA2" ( or LeaderA4, etc. ) to be simply "LeaderA". Otherwise the code is a direct copy of the original module, just grouped differently ( and hopefully more sanely ).